spec/models/ai_persona_spec.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+RSpec.describe AiPersona do
+  it "does not leak caches between sites" do
+    AiPersona.create!(
+      name: "pun_bot",
+      description: "you write puns",
+      system_prompt: "you are pun bot",
+      commands: ["ImageCommand"],
+      allowed_group_ids: [Group::AUTO_GROUPS[:trust_level_0]],
+    )
+
+    AiPersona.all_personas
+
+    expect(AiPersona.persona_cache[:value].length).to eq(1)
+    RailsMultisite::ConnectionManagement.stubs(:current_db) { "abc" }
```

Comment:
For this we have `test_multisite_connection`:

https://github.com/discourse/discourse/blob/813278717111a8a545d64330decdceb0be6a0006/spec/multisite/site_settings_spec.rb#L15-L19

Just note that the `"second"` is important, it only works with a few predefined values.